src/systems/drawing-menu-system.js

```diff
@@ -0,0 +1,200 @@
+  constructor(sceneEl) {
+    this.sceneEl = sceneEl;
+
+    this.sceneEl.addEventListener("stateremoved", this.hideMenus.bind(this));
+
+    this.lastIntersection = new THREE.Vector3();
+
+    this.drawingMenus = [];
+    this.setDirty = this.setDirty.bind(this);
+    this.dirty = true;
+
+    // TODO: Use the MutationRecords passed into the callback function to determine added/removed nodes!
+    this.observer = new MutationObserver(this.setDirty);
+
+    waitForDOMContentLoaded().then(() => {
+      this.cursorControllers = document.querySelectorAll("[cursor-controller]");
+      this.camera = this.sceneEl.querySelector("#avatar-pov-node");
+      this.observer.observe(this.sceneEl, { childList: true, attributes: true, subtree: true });
+      this.sceneEl.addEventListener("object3dset", this.setDirty.bind(this));
+      this.sceneEl.addEventListener("object3dremove", this.setDirty.bind(this));
+    });
+  }
+
+  tick() {
+    if (!this.cursorControllers || this.cursorControllers.length === 0 || !this.camera) return;
+
+    if (this.sceneEl.is("frozen")) {
+      if (this.dirty) {
+        this.populateEntities(this.targets);
+        this.dirty = false;
+      }
+
+      const hovered = this.getHovered();
+      if (hovered) {
+        for (let i = 0; i < this.cursorControllers.length; i++) {
+          if (this.cursorControllers[i].components["cursor-controller"].intersection) {
+            const intersectionPoint = this.cursorControllers[i].components["cursor-controller"].intersection.point;
+            this.showMenu(hovered, intersectionPoint);
+          }
+        }
+      }
+      this.handleButtons();
+      this.scaleMenus();
+    }
+  }
+
+  getMenuScale(distance) {
+    if (distance <= 1.5) {
+      return 0.5;
+    }
+    return Math.min(3, distance - 1);
+  }
+
+  scaleMenus = (() => {
+    const cameraWorldPos = new THREE.Vector3();
+    return function() {
+      if (!this.drawingMenus || this.drawingMenus.length === 0) return;
+
+      this.camera.object3D.getWorldPosition(cameraWorldPos);
+
+      for (let i = 0; i < this.drawingMenus.length; i++) {
+        const menu = this.drawingMenus[i];
+        if (menu.object3D.visible && !menu.getAttribute("animation__show")) {
+          const dist = cameraWorldPos.distanceTo(menu.object3D.position);
+          menu.object3D.scale.setScalar(this.getMenuScale(dist));
+          menu.object3D.matrixNeedsUpdate = true;
+        }
+      }
+    };
+  })();
+
+  handleButtons = (() => {
+    const remotes = ["rightRemote", "leftRemote"];
+    return function() {
+      const interaction = AFRAME.scenes[0].systems.interaction;
+      const userinput = AFRAME.scenes[0].systems.userinput;
+
+      for (let i = 0; i < remotes.length; i++) {
+        const remote = remotes[i];
+        if (interaction.state[remote]) {
+          const hovered = interaction.state[remote].hovered;
+          if (
+            hovered &&
+            interaction.options[remote] &&
```

Comment:
When does `interaction.options[remote]` yield a falsey value?

Reply:
This was written before the left handed changes got merged